src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,163 @@
+            entry.previous.next = newEntry;
+            entry.previous = newEntry;
+            if (index == size - 1) {
+                newEntry.next = entry;
+            } else {
+                newEntry.next.previous = newEntry;
+            }
+        }
+        ++size;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        if (list == null || list.isEmpty()) {
+            return false;
+        }
+        for (T t : list) {
+            add(t);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Entry<T> entry = findEntryByIndex(index);
+        return entry.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Entry<T> entry = findEntryByIndex(index);
+        T element = entry.item;
+        entry.item = value;
+        return element;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        if (index == 0) {
+            T element = first.item;
+            removeFirstElement();
+            return element;
+        }
+        Entry<T> entry = findEntryByIndex(index);
+        removeElement(entry);
+        return entry.item;
     }
 
     @Override
     public boolean remove(T t) {
+        Entry<T> entry = first;
+        for (int i = 0; i < size; i++) {
+            if ((entry.item == t) || (entry.item != null && entry.item.equals(t))) {
+                if (i == 0) {
+                    removeFirstElement();
+                    return true;
+                }
+                removeElement(entry);
+                return true;
+            }
+            entry = first.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Incorrect index!");
+        }
+    }
+
+    private Entry<T> findEntryByIndex(int index) {
+        Entry<T> entry = first;
+        for (int i = 0; i < index; i++) {
+            entry = entry.next;
+        }
+        return entry;
+    }
+
+    private void removeFirstElement() {
+        if (size == 1) {
+            first = null;
+            last = null;
+        } else {
+            first.next.previous = null;
+            first = first.next;
+        }
+        --size;
+    }
+
+    private void removeElement(Entry<T> entry) {
+        if (entry.equals(last)) {
+            last.previous.next = null;
+            last = entry.previous;
+        } else {
+            entry.previous.next = entry.next;
+            entry.next.previous = entry.previous;
+        }
```

Comment:
here I would suggest to create two local variables that point to `entry.previous` and `entry.next`
then write the unlinking logic for `entry.previous` (add check if your `entry` is `first` here)
and the unlinking logic for `entry.next` (add check if your `entry` is `last` here`)